[6주차] Restaurant router

hanna244 opened the pull request:
안녕하세요 잘부탁드립니다!

hanna244 committed:
react router 설치

hanna244 committed:
RestaurantsPage 추가

hanna244 committed:
react router 세팅

hanna244 committed:
HomePage test

hanna244 committed:
HomePage implement

hanna244 committed:
App test

KJoobin commented on src/App.test.jsx:
```diff
@@ -2,33 +2,29 @@ import React from 'react';
+
+    expect
+  });
+});
```
⛔️ (No newline at end of file) 이 있네요 :)

KJoobin commented on src/HomePage.test.jsx:
```diff
@@ -0,0 +1,16 @@
+
+import HomePage from './HomePage';
+
+describe('home page', () => {
+  it('home page', () => {
```
describe - context - it 순으로 읽었을때 어색하지 않게 적어주시면 좋을거 같습니다 :)

KJoobin reviewed (changes requested):
TDD 싸이클을 잘 진행해주신거 같습니다 :) 

lint 를 조금더 신경써주시면 좋을거같아요!! 남은 과제도 화이팅 👏👏👏

hanna244 committed:
에러 대응을 위한 코드 수정 (해결 안됨)
- 에러: TypeError: Cannot read properties of undefined (reading 'map')
- useSelector 추가

hanna244 committed:
App test - import Page components

hanna244 committed:
App refactor - renderApp 유틸 함수 만들기

hanna244 committed:
AboutPage test

hanna244 committed:
AboutPage implement

hanna244 committed:
App Route 설정

KJoobin commented on src/AboutPage.test.jsx:
```diff
@@ -0,0 +1,17 @@
+        <MemoryRouter>
+          <AboutPage />
+        </MemoryRouter>,
+      );
+      expect(container).toHaveTextContent('About');
```
한줄 띄어주시면 더 좋을거같아요 :)

KJoobin commented on src/App.test.jsx:
```diff
@@ -2,33 +2,61 @@ import React from 'react';
+    });
+  });
+
+  // context('with path /restaurants', () => {
+  //   it('render RestaurantsPage', () => {
+  //     const { container } = renderApp({ padth: '/restaurants' });
+
+  //     expect(container).toHaveTextContent('강원도');
+  //   });
+  // });
```
해당 주석은 에러때문에 만드신건가요??
(+ 에러는 잘 해결하셨을까요?)

KJoobin reviewed (commented):
바쁘실텐데 꾸준히 진행해주시는 모습이 보기 좋습니다 :) 👍

1일 1PR 을 꼭 완성해봐요!!

막히는 부분이 있다면 편하게 DM 주셔도 괜찮습니다 :)

hanna244 committed:
에러 해결 MemoryRouter 추가 및 오타 수정

hanna244 committed:
RestaurantDetailPage test

hanna244 committed:
refactor - 오타 수정 및 lint 경고 대응

hanna244 committed:
App test

KJoobin commented on src/App.jsx:
```diff
@@ -11,6 +12,7 @@ export default function App() {
         <Route path="/" element={<HomePage />} />
         <Route path="/about" element={<AboutPage />} />
         <Route path="/restaurants" element={<RestaurantsPage />} />
+        <Route path="/restaurant-detail" element={<RestaurantDetailPage />} />
```
route path 는 restful 하게 작성해주시면 좋습니다 :)

KJoobin commented on src/RestaurantDetailPage.jsx:
```diff
@@ -0,0 +1,33 @@
+export default function RestaurantDetailPage() {
+  const {
+    name, address, menuItems, reviews,
+  } = useSelector(get('restaurantDetail'));
```
restaurantDetail 은 조금 어색한거 같아요..!

hanna244 committed:
test 에러 대응

hanna244 committed:
RestaurantDetail action 및 reducer 추가

hanna244 committed:
loadRestaurantDetail 로직 수정

KJoobin commented on src/RestaurantDetailPage.jsx:
```diff
@@ -30,4 +30,4 @@ export default function RestaurantDetailPage() {
     </div>
   );
-}
+}
```
커밋을 하기전에 lint 를 돌리는 습관을 들이시면 좋을거 같습니다 :)

KJoobin commented on src/services/api.js:
```diff
@@ -13,8 +13,16 @@ export async function fetchCategories() {
 export async function fetchRestaurants({ regionName, categoryId }) {
-  const url = 'https://eatgo-customer-api.ahastudio.com/restaurants'
-    + `?region=${regionName}&category=${categoryId}`;
+  const url =
+    'https://eatgo-customer-api.ahastudio.com/restaurants' +
+    `?region=${regionName}&category=${categoryId}`;
```
중복되는곳이 보이는거같아요 :)

KJoobin reviewed (commented):
TDD 사이클을 돌면서 진행중이신가요?? ;)

hanna244 committed:
selectRestaurant action test

hanna244 committed:
RestaurantsContainer dispatch test

hanna244 committed:
RestaurantsContainer dispatch 구현

hanna244 committed:
eslint 대응

hanna244 committed:
RestaurantDetailPage dispatch test

hanna244 committed:
RestaurantDetailPage disapatch 구현

hanna244 committed:
refactor: 레스토랑 상세 페이지에 로딩중 마크업 추가

hanna244 committed:
refactor - restoreAllMocks을 사용한 모킹한 mock 정리

hanna244 committed:
refactor - App test

KJoobin commented on src/RestaurantsContainer.jsx:
```diff
@@ -1,17 +1,26 @@
+  const dispatch = useDispatch();
   const restaurants = useSelector(get('restaurants'));
 
+  function handleClickRestaurant(e) {
+    dispatch(selectRestaurant(e.target.id));
+  }
```
router 을 RESTful 하게 작성해주시면 더 좋을거같아요!
`restaurants` `restaurants/{restaurant.id}` 그러면 `handleClickRestaurant` 메소드는 필요없겠죠?? :)

KJoobin commented on src/actions.test.js:
```diff
@@ -58,7 +62,7 @@ describe('actions', () => {
       });
 
-      it('does\'nt run any actions', async () => {
+      it("does'nt run any actions", async () => {
```
```suggestion
      it("doesn't run any actions", async () => {
```

KJoobin commented on src/reducer.test.js:
```diff
@@ -108,4 +106,16 @@ describe('reducer', () => {
     });
   });
+
+  describe('', () => {
```
```suggestion
  describe('selectRestaurant', () => {
```

KJoobin commented:
해설 영상을 보고 남은 부분도 채워주세요 :)
